dev/tools/bin/generate_gradle_lockfiles.dart

```diff
@@ -8,16 +8,36 @@
 
 import 'dart:io';
 
+import 'package:args/args.dart';
 import 'package:file/file.dart';
 import 'package:file/local.dart';
 import 'package:path/path.dart' as path;
 
 void main(List<String> arguments) {
-  print(
-    "Usage: find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart\n"
-    'If you would rather enter the files manually, just run `dart dev/tools/bin/generate_gradle_lockfiles.dart`,\n'
-    "enter the absolute paths to the app's android directory, then press CTRL-D.\n"
-  );
+  const String usageMessage = "Usage: find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart\n"
+      'If you would rather enter the files manually, just run `dart dev/tools/bin/generate_gradle_lockfiles.dart`,\n'
+      "enter the absolute paths to the app's android directory, then press CTRL-D.\n"
+      "If you don't wish to re-generate the settings.gradle, build.gradle, and gradle-wrapper.properties files,\n"
```

Comment:
Shouldn't this be the opposite?

Reply:
Hmm as in default behavior, or just the documentation? 

I think the documentation is correct as written (the default is to re-generate, the flag makes it not re-generate)

Reply:
Ahh ok, you're correct. I would recommend instead naming the flag `gradle-generation`, with the argument `defaultsTo: true`. This will implicitly create a `--no-gradle-generation` flag that will allow opt-out.

As for default behavior, I'm unclear. When would I want to regenerate these vs not generate them?

Reply:
> Ahh ok, you're correct. I would recommend instead naming the flag `gradle-generation`, with the argument `defaultsTo: true`. This will implicitly create a `--no-gradle-generation` flag that will allow opt-out.

I did not know this! Changed the flag and gave it true default (though I left the documentation as is because it is still true that you pass the `--no-gradle-generation` flag to not generate.

> As for default behavior, I'm unclear. When would I want to regenerate these vs not generate them?

The main use case is batch updating the gradle files to new versions of Gradle/AGP/kotlin. I think the logic was originally included to add the lines:
```
buildscript {
    dependencyLocking {
        lockFile = file("${rootProject.projectDir}/buildscript-gradle.lockfile")
        lockAllConfigurations()
    }
}
``` 
to build.gradle files that didn't have them, but after that has been included once I think re-generating the gradle files is actually pretty orthogonal to the task of updating the lockfiles.

But in spite of that I still sort of soft lean towards defaulting to true because it doesn't make a big difference and it keeps the behavior in line with many breadcrumb comments I've made in the past

Reply:
ok, i'll leave it up to you (as probably the primary user). if/when I wire this up to the auto-roller, I'll be sure to opt out of regenerating those files.